src/blob/core.rs

```diff
@@ -180,7 +180,18 @@ impl Blob {
         debug!("blob write record offset: {}", *offset);
         record.set_offset(*offset)?;
         let buf = record.to_raw()?;
-        let bytes_written = self.file.write_append(&buf).await? as u64;
+        let bytes_written = self
+            .file
+            .write_append(&buf)
+            .await
+            .map_err(|e| -> anyhow::Error {
+                match e.kind() {
+                    kind if kind == IOErrorKind::Other || kind == IOErrorKind::NotFound => {
+                        Error::file_unavailable(kind).into()
```

Comment:
`?` performs `into()` conversion under the hood, are you sure it is necessary here?

Reply:
I convert some of OS errors which may occur due to disk crash to Pearl error which indicates that. There are different errors in branch so I need to convert error to `anyhow::Error` inside of closure, and that's before `?` operator action.